docs/source/data/data_catalog.md

```diff
@@ -1,31 +1,34 @@
+```yaml
+companies:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/companies.csv
 
-Built-in functionality for `conf/local/` to overwrite `conf/base/` is [described in the documentation about configuration](../configuration/configuration_basics.md). This means that a dataset called `cars` could exist in the `catalog.yml` files in `conf/base/` and `conf/local/`. In code, in `src`, you would only call a dataset named `cars` and Kedro would detect which definition of `cars` dataset to use to run your pipeline - `cars` definition from `conf/local/catalog.yml` would take precedence in this case.
+reviews:
+  type: pandas.CSVDataSet
+  filepath: data/01_raw/reviews.csv
 
-The Data Catalog also works with the `credentials.yml` file in `conf/local/`, allowing you to specify usernames and passwords required to load certain datasets.
+shuttles:
+  type: pandas.ExcelDataSet
+  filepath: data/01_raw/shuttles.xlsx
+  load_args:
+    engine: openpyxl # Use modern Excel engine (the default since Kedro 0.18.0)
+```
+### Dataset `type`
 
-You can define a Data Catalog in two ways - through YAML configuration, or programmatically using an API. Both methods allow you to specify:
+Kedro offers a range of datasets, including CSV, Excel, Parquet, Feather, HDF5, JSON, Pickle, SQL Tables, SQL Queries, Spark DataFrames and more. They are supported with the APIs of pandas, spark, networkx, matplotlib, yaml and more.
 
- - Dataset name
- - Dataset type
- - Location of the dataset using `fsspec`, detailed in the next section
- - Credentials needed to access the dataset
- - Load and saving arguments
- - Whether you want a [dataset or ML model to be versioned](kedro_io.md#versioning) when you run your data pipeline
+[The `kedro-datasets` package documentation](/kedro_datasets) contains a comprehensive list of all available file types.
 
-## Specify the location of the dataset
+### Dataset `filepath`
 
-Kedro relies on [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to read and save data from a variety of data stores including local file systems, network file systems, cloud object stores, and Hadoop. When specifying a storage location in `filepath:`, you should provide a URL using the general form `protocol://path/to/data`.  If no protocol is provided, the local file system is assumed (same as ``file://``).
+Kedro relies on [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to read and save data from a variety of data stores including local file systems, network file systems, cloud object stores, and Hadoop. When specifying a storage location in `filepath:`, you should provide a URL using the general form `protocol://path/to/data`.  If no protocol is provided, the local file system is assumed (which is the same as ``file://``).
```

Comment:
Worth mentioning (maybe a "note" admonition) that there's some magical mangling with `filepath:` and `file:` properties?

https://github.com/kedro-org/kedro/issues/2819#issuecomment-1669748465

(which means that relative _local_ paths are relative with respect to the _root_ of the project)

Reply:
Possibly but I'm not sure what to say. Maybe just another sentence so it reads

```suggestion
Kedro relies on [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to read and save data from a variety of data stores including local file systems, network file systems, cloud object stores, and Hadoop. When specifying a storage location in `filepath:`, you should provide a URL using the general form `protocol://path/to/data`.  If no protocol is provided, the local file system is assumed (which is the same as ``file://``). Note that relative local paths are interpreted as relative to the root of the project.
```

I don't want to get too hardcore in this page as this is the intro/non-advanced user information. Is there anywhere else we could put this, such as in the API ref?

Reply:
Yeah you're right, let's keep this document like this, and I opened another issue to document this "magic" behavior.

https://github.com/kedro-org/kedro/issues/2942